code/modules/surgery/organs/liver.dm

```diff
@@ -0,0 +1,27 @@
+/obj/item/organ/liver
+	name = "liver"
+	icon_state = "liver"
+	origin_tech = "biotech=5"
+	w_class = WEIGHT_CLASS_NORMAL
+	zone = "chest"
+	slot = "liver"
+	desc = "Goes great with a nice chianti and some fava beans."
+	var/damage = 0 //liver damage, 0 is no damage, 100 causes liver failure
+	var/alcohol_tolerance = ALCOHOL_RATE
+	var/failing //is this liver failing?
+
+/obj/item/organ/liver/on_life()
+	var/mob/living/carbon/C = owner
+
+	//slowly heal liver damage
+	if(damage > 0)
+		damage -= 0.01//in most cases the pure toxin damage from too much alcohol will kill you first.
+
+	if(iscarbon(C))
+		if(C.reagents)
+			C.reagents.metabolize(C, can_overdose=1)

```

Comment:
You killed off the ability for species to handle specific chemicals.

Also, as with the stomach, it would be interesting to have the stomach handle species reagent process uniques on itself.

Meaning if you gave a plant person's stomach to a regular individual, they'd take extra tox from plant-b-gone.